strawberry_django_plus/utils/query.py

```diff
@@ -1,24 +1,34 @@
 import functools
-from typing import List, Optional, Set, Type, TypeVar
+from typing import TYPE_CHECKING, List, Optional, Set, Type, TypeVar, cast
 
-from django.contrib.auth import get_user_model
-from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import FieldDoesNotExist
 from django.db.models import Exists, F, Model, Q, QuerySet
 from django.db.models.functions import Cast
 from strawberry_django.utils import is_async
 
 from .typing import TypeOrIterable, UserType
 
+if TYPE_CHECKING:
+    from django.contrib.auth.models import AbstractUser
+try:
+    from django.contrib.contenttypes.models import ContentType
+
+    has_contenttypes = True
+except (ImportError, RuntimeError):
+    # for pyright introduce an extra variable
+    # ContentType = None makes ContentType an invalid type
+    has_contenttypes = False
```

Comment:
Ditto here. Instead of `has_contenttypes` set it to `None` and check if it is `None` where you are using (like I suggested in another comment)

Reply:
done with a seperate type import